### PR TITLE
Update pydantic taxprofiler

### PR DESCRIPTION
**Pull request by henrikstranneheim**
### Added

-

### Changed

-

### Fixed

-

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b update-pydantic-taxprofiler
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

**Comment by henrikstranneheim** [comment]
### cg workflow taxprofiler start, 👍 
```
henrik.stranneheim@hasta:~] [S_base] $ cg workflow taxprofiler start richurchin --dry-run
Called undefined __pydantic_serializer__ on HousekeeperAPI, please wrap
Called undefined __dataclass_fields__ on HousekeeperAPI, please wrap
Called undefined __pydantic_serializer__ on HousekeeperAPI, please wrap
Called undefined __dataclass_fields__ on HousekeeperAPI, please wrap
Called undefined __pydantic_serializer__ on HousekeeperAPI, please wrap
Called undefined __dataclass_fields__ on HousekeeperAPI, please wrap
Called undefined __pydantic_serializer__ on HousekeeperAPI, please wrap
Called undefined __dataclass_fields__ on HousekeeperAPI, please wrap
Called undefined __pydantic_serializer__ on HousekeeperAPI, please wrap
Called undefined __dataclass_fields__ on HousekeeperAPI, please wrap
Called undefined __pydantic_serializer__ on HousekeeperAPI, please wrap
Called undefined __dataclass_fields__ on HousekeeperAPI, please wrap
Called undefined __pydantic_serializer__ on HousekeeperAPI, please wrap
Called undefined __dataclass_fields__ on HousekeeperAPI, please wrap
Called undefined __pydantic_serializer__ on HousekeeperAPI, please wrap
Called undefined __dataclass_fields__ on HousekeeperAPI, please wrap
Called undefined __pydantic_serializer__ on HousekeeperAPI, please wrap
Called undefined __dataclass_fields__ on HousekeeperAPI, please wrap
Starting analysis for richurchin
Case richurchin exists in Status DB
Check if pending compression file exists
/home/proj/stage/housekeeper-bundles/ACC10629A1/2022-10-30/HMLN5DSX3_ACC10629A1_S93_L001.crunchy.pending.txt does not exist
Compression/decompression is not running
Check if FASTQ pair exists
Check if pending compression file exists
/home/proj/stage/housekeeper-bundles/ACC10629A1/2022-10-30/HMLN5DSX3_ACC10629A1_S93_L004.crunchy.pending.txt does not exist
Compression/decompression is not running
Check if FASTQ pair exists
Check if pending compression file exists
/home/proj/stage/housekeeper-bundles/ACC10629A1/2022-10-30/HMLN5DSX3_ACC10629A1_S93_L002.crunchy.pending.txt does not exist
Compression/decompression is not running
Check if FASTQ pair exists
Check if pending compression file exists
/home/proj/stage/housekeeper-bundles/ACC10629A1/2022-10-30/HMLN5DSX3_ACC10629A1_S93_L003.crunchy.pending.txt does not exist
Compression/decompression is not running
Check if FASTQ pair exists
Check if pending compression file exists
/home/proj/stage/housekeeper-bundles/ACC10629A2/2022-10-30/HMLN5DSX3_ACC10629A2_S97_L001.crunchy.pending.txt does not exist
Compression/decompression is not running
Check if FASTQ pair exists
Check if pending compression file exists
/home/proj/stage/housekeeper-bundles/ACC10629A2/2022-10-30/HMLN5DSX3_ACC10629A2_S97_L002.crunchy.pending.txt does not exist
Compression/decompression is not running
Check if FASTQ pair exists
Check if pending compression file exists
/home/proj/stage/housekeeper-bundles/ACC10629A2/2022-10-30/HMLN5DSX3_ACC10629A2_S97_L004.crunchy.pending.txt does not exist
Compression/decompression is not running
Check if FASTQ pair exists
Check if pending compression file exists
/home/proj/stage/housekeeper-bundles/ACC10629A2/2022-10-30/HMLN5DSX3_ACC10629A2_S97_L003.crunchy.pending.txt does not exist
Compression/decompression is not running
Check if FASTQ pair exists
Check if pending compression file exists
/home/proj/stage/housekeeper-bundles/ACC10629A3/2022-10-30/HMLN5DSX3_ACC10629A3_S101_L002.crunchy.pending.txt does not exist
Compression/decompression is not running
Check if FASTQ pair exists
Check if pending compression file exists
/home/proj/stage/housekeeper-bundles/ACC10629A3/2022-10-30/HMLN5DSX3_ACC10629A3_S101_L001.crunchy.pending.txt does not exist
Compression/decompression is not running
Check if FASTQ pair exists
Check if pending compression file exists
/home/proj/stage/housekeeper-bundles/ACC10629A3/2022-10-30/HMLN5DSX3_ACC10629A3_S101_L003.crunchy.pending.txt does not exist
Compression/decompression is not running
Check if FASTQ pair exists
Check if pending compression file exists
/home/proj/stage/housekeeper-bundles/ACC10629A3/2022-10-30/HMLN5DSX3_ACC10629A3_S101_L004.crunchy.pending.txt does not exist
Compression/decompression is not running
Check if FASTQ pair exists
Check if pending compression file exists
/home/proj/stage/housekeeper-bundles/ACC10629A1/2022-10-30/HMLN5DSX3_ACC10629A1_S93_L001.crunchy.pending.txt does not exist
Compression/decompression is not running
Check if pending compression file exists
/home/proj/stage/housekeeper-bundles/ACC10629A1/2022-10-30/HMLN5DSX3_ACC10629A1_S93_L004.crunchy.pending.txt does not exist
Compression/decompression is not running
Check if pending compression file exists
/home/proj/stage/housekeeper-bundles/ACC10629A1/2022-10-30/HMLN5DSX3_ACC10629A1_S93_L002.crunchy.pending.txt does not exist
Compression/decompression is not running
Check if pending compression file exists
/home/proj/stage/housekeeper-bundles/ACC10629A1/2022-10-30/HMLN5DSX3_ACC10629A1_S93_L003.crunchy.pending.txt does not exist
Compression/decompression is not running
Check if pending compression file exists
/home/proj/stage/housekeeper-bundles/ACC10629A2/2022-10-30/HMLN5DSX3_ACC10629A2_S97_L001.crunchy.pending.txt does not exist
Compression/decompression is not running
Check if pending compression file exists
/home/proj/stage/housekeeper-bundles/ACC10629A2/2022-10-30/HMLN5DSX3_ACC10629A2_S97_L002.crunchy.pending.txt does not exist
Compression/decompression is not running
Check if pending compression file exists
/home/proj/stage/housekeeper-bundles/ACC10629A2/2022-10-30/HMLN5DSX3_ACC10629A2_S97_L004.crunchy.pending.txt does not exist
Compression/decompression is not running
Check if pending compression file exists
/home/proj/stage/housekeeper-bundles/ACC10629A2/2022-10-30/HMLN5DSX3_ACC10629A2_S97_L003.crunchy.pending.txt does not exist
Compression/decompression is not running
Check if pending compression file exists
/home/proj/stage/housekeeper-bundles/ACC10629A3/2022-10-30/HMLN5DSX3_ACC10629A3_S101_L002.crunchy.pending.txt does not exist
Compression/decompression is not running
Check if pending compression file exists
/home/proj/stage/housekeeper-bundles/ACC10629A3/2022-10-30/HMLN5DSX3_ACC10629A3_S101_L001.crunchy.pending.txt does not exist
Compression/decompression is not running
Check if pending compression file exists
/home/proj/stage/housekeeper-bundles/ACC10629A3/2022-10-30/HMLN5DSX3_ACC10629A3_S101_L003.crunchy.pending.txt does not exist
Compression/decompression is not running
Check if pending compression file exists
/home/proj/stage/housekeeper-bundles/ACC10629A3/2022-10-30/HMLN5DSX3_ACC10629A3_S101_L004.crunchy.pending.txt does not exist
Compression/decompression is not running
Both /home/proj/stage/housekeeper-bundles/ACC10629A1/2022-10-30/HMLN5DSX3_ACC10629A1_S93_L001_R1_001.fastq.gz and /home/proj/stage/housekeeper-bundles/ACC10629A1/2022-10-30/HMLN5DSX3_ACC10629A1_S93_L001_R2_001.fastq.gz from sample ACC10629A1 are already in Housekeeper
Both /home/proj/stage/housekeeper-bundles/ACC10629A1/2022-10-30/HMLN5DSX3_ACC10629A1_S93_L004_R1_001.fastq.gz and /home/proj/stage/housekeeper-bundles/ACC10629A1/2022-10-30/HMLN5DSX3_ACC10629A1_S93_L004_R2_001.fastq.gz from sample ACC10629A1 are already in Housekeeper
Both /home/proj/stage/housekeeper-bundles/ACC10629A1/2022-10-30/HMLN5DSX3_ACC10629A1_S93_L002_R1_001.fastq.gz and /home/proj/stage/housekeeper-bundles/ACC10629A1/2022-10-30/HMLN5DSX3_ACC10629A1_S93_L002_R2_001.fastq.gz from sample ACC10629A1 are already in Housekeeper
Both /home/proj/stage/housekeeper-bundles/ACC10629A1/2022-10-30/HMLN5DSX3_ACC10629A1_S93_L003_R1_001.fastq.gz and /home/proj/stage/housekeeper-bundles/ACC10629A1/2022-10-30/HMLN5DSX3_ACC10629A1_S93_L003_R2_001.fastq.gz from sample ACC10629A1 are already in Housekeeper
Both /home/proj/stage/housekeeper-bundles/ACC10629A2/2022-10-30/HMLN5DSX3_ACC10629A2_S97_L001_R1_001.fastq.gz and /home/proj/stage/housekeeper-bundles/ACC10629A2/2022-10-30/HMLN5DSX3_ACC10629A2_S97_L001_R2_001.fastq.gz from sample ACC10629A2 are already in Housekeeper
Both /home/proj/stage/housekeeper-bundles/ACC10629A2/2022-10-30/HMLN5DSX3_ACC10629A2_S97_L002_R1_001.fastq.gz and /home/proj/stage/housekeeper-bundles/ACC10629A2/2022-10-30/HMLN5DSX3_ACC10629A2_S97_L002_R2_001.fastq.gz from sample ACC10629A2 are already in Housekeeper
Both /home/proj/stage/housekeeper-bundles/ACC10629A2/2022-10-30/HMLN5DSX3_ACC10629A2_S97_L004_R1_001.fastq.gz and /home/proj/stage/housekeeper-bundles/ACC10629A2/2022-10-30/HMLN5DSX3_ACC10629A2_S97_L004_R2_001.fastq.gz from sample ACC10629A2 are already in Housekeeper
Both /home/proj/stage/housekeeper-bundles/ACC10629A2/2022-10-30/HMLN5DSX3_ACC10629A2_S97_L003_R1_001.fastq.gz and /home/proj/stage/housekeeper-bundles/ACC10629A2/2022-10-30/HMLN5DSX3_ACC10629A2_S97_L003_R2_001.fastq.gz from sample ACC10629A2 are already in Housekeeper
Both /home/proj/stage/housekeeper-bundles/ACC10629A3/2022-10-30/HMLN5DSX3_ACC10629A3_S101_L002_R1_001.fastq.gz and /home/proj/stage/housekeeper-bundles/ACC10629A3/2022-10-30/HMLN5DSX3_ACC10629A3_S101_L002_R2_001.fastq.gz from sample ACC10629A3 are already in Housekeeper
Both /home/proj/stage/housekeeper-bundles/ACC10629A3/2022-10-30/HMLN5DSX3_ACC10629A3_S101_L001_R1_001.fastq.gz and /home/proj/stage/housekeeper-bundles/ACC10629A3/2022-10-30/HMLN5DSX3_ACC10629A3_S101_L001_R2_001.fastq.gz from sample ACC10629A3 are already in Housekeeper
Both /home/proj/stage/housekeeper-bundles/ACC10629A3/2022-10-30/HMLN5DSX3_ACC10629A3_S101_L003_R1_001.fastq.gz and /home/proj/stage/housekeeper-bundles/ACC10629A3/2022-10-30/HMLN5DSX3_ACC10629A3_S101_L003_R2_001.fastq.gz from sample ACC10629A3 are already in Housekeeper
Both /home/proj/stage/housekeeper-bundles/ACC10629A3/2022-10-30/HMLN5DSX3_ACC10629A3_S101_L004_R1_001.fastq.gz and /home/proj/stage/housekeeper-bundles/ACC10629A3/2022-10-30/HMLN5DSX3_ACC10629A3_S101_L004_R2_001.fastq.gz from sample ACC10629A3 are already in Housekeeper
Check if pending compression file exists
/home/proj/stage/housekeeper-bundles/ACC10629A1/2022-10-30/HMLN5DSX3_ACC10629A1_S93_L001.crunchy.pending.txt does not exist
Compression/decompression is not running
Check if FASTQ pair exists
Check if pending compression file exists
/home/proj/stage/housekeeper-bundles/ACC10629A1/2022-10-30/HMLN5DSX3_ACC10629A1_S93_L004.crunchy.pending.txt does not exist
Compression/decompression is not running
Check if FASTQ pair exists
Check if pending compression file exists
/home/proj/stage/housekeeper-bundles/ACC10629A1/2022-10-30/HMLN5DSX3_ACC10629A1_S93_L002.crunchy.pending.txt does not exist
Compression/decompression is not running
Check if FASTQ pair exists
Check if pending compression file exists
/home/proj/stage/housekeeper-bundles/ACC10629A1/2022-10-30/HMLN5DSX3_ACC10629A1_S93_L003.crunchy.pending.txt does not exist
Compression/decompression is not running
Check if FASTQ pair exists
Check if pending compression file exists
/home/proj/stage/housekeeper-bundles/ACC10629A2/2022-10-30/HMLN5DSX3_ACC10629A2_S97_L001.crunchy.pending.txt does not exist
Compression/decompression is not running
Check if FASTQ pair exists
Check if pending compression file exists
/home/proj/stage/housekeeper-bundles/ACC10629A2/2022-10-30/HMLN5DSX3_ACC10629A2_S97_L002.crunchy.pending.txt does not exist
Compression/decompression is not running
Check if FASTQ pair exists
Check if pending compression file exists
/home/proj/stage/housekeeper-bundles/ACC10629A2/2022-10-30/HMLN5DSX3_ACC10629A2_S97_L004.crunchy.pending.txt does not exist
Compression/decompression is not running
Check if FASTQ pair exists
Check if pending compression file exists
/home/proj/stage/housekeeper-bundles/ACC10629A2/2022-10-30/HMLN5DSX3_ACC10629A2_S97_L003.crunchy.pending.txt does not exist
Compression/decompression is not running
Check if FASTQ pair exists
Check if pending compression file exists
/home/proj/stage/housekeeper-bundles/ACC10629A3/2022-10-30/HMLN5DSX3_ACC10629A3_S101_L002.crunchy.pending.txt does not exist
Compression/decompression is not running
Check if FASTQ pair exists
Check if pending compression file exists
/home/proj/stage/housekeeper-bundles/ACC10629A3/2022-10-30/HMLN5DSX3_ACC10629A3_S101_L001.crunchy.pending.txt does not exist
Compression/decompression is not running
Check if FASTQ pair exists
Check if pending compression file exists
/home/proj/stage/housekeeper-bundles/ACC10629A3/2022-10-30/HMLN5DSX3_ACC10629A3_S101_L003.crunchy.pending.txt does not exist
Compression/decompression is not running
Check if FASTQ pair exists
Check if pending compression file exists
/home/proj/stage/housekeeper-bundles/ACC10629A3/2022-10-30/HMLN5DSX3_ACC10629A3_S101_L004.crunchy.pending.txt does not exist
Compression/decompression is not running
Check if FASTQ pair exists
Check if pending compression file exists
/home/proj/stage/housekeeper-bundles/ACC10629A1/2022-10-30/HMLN5DSX3_ACC10629A1_S93_L001.crunchy.pending.txt does not exist
Compression/decompression is not running
Check if pending compression file exists
/home/proj/stage/housekeeper-bundles/ACC10629A1/2022-10-30/HMLN5DSX3_ACC10629A1_S93_L004.crunchy.pending.txt does not exist
Compression/decompression is not running
Check if pending compression file exists
/home/proj/stage/housekeeper-bundles/ACC10629A1/2022-10-30/HMLN5DSX3_ACC10629A1_S93_L002.crunchy.pending.txt does not exist
Compression/decompression is not running
Check if pending compression file exists
/home/proj/stage/housekeeper-bundles/ACC10629A1/2022-10-30/HMLN5DSX3_ACC10629A1_S93_L003.crunchy.pending.txt does not exist
Compression/decompression is not running
Check if pending compression file exists
/home/proj/stage/housekeeper-bundles/ACC10629A2/2022-10-30/HMLN5DSX3_ACC10629A2_S97_L001.crunchy.pending.txt does not exist
Compression/decompression is not running
Check if pending compression file exists
/home/proj/stage/housekeeper-bundles/ACC10629A2/2022-10-30/HMLN5DSX3_ACC10629A2_S97_L002.crunchy.pending.txt does not exist
Compression/decompression is not running
Check if pending compression file exists
/home/proj/stage/housekeeper-bundles/ACC10629A2/2022-10-30/HMLN5DSX3_ACC10629A2_S97_L004.crunchy.pending.txt does not exist
Compression/decompression is not running
Check if pending compression file exists
/home/proj/stage/housekeeper-bundles/ACC10629A2/2022-10-30/HMLN5DSX3_ACC10629A2_S97_L003.crunchy.pending.txt does not exist
Compression/decompression is not running
Check if pending compression file exists
/home/proj/stage/housekeeper-bundles/ACC10629A3/2022-10-30/HMLN5DSX3_ACC10629A3_S101_L002.crunchy.pending.txt does not exist
Compression/decompression is not running
Check if pending compression file exists
/home/proj/stage/housekeeper-bundles/ACC10629A3/2022-10-30/HMLN5DSX3_ACC10629A3_S101_L001.crunchy.pending.txt does not exist
Compression/decompression is not running
Check if pending compression file exists
/home/proj/stage/housekeeper-bundles/ACC10629A3/2022-10-30/HMLN5DSX3_ACC10629A3_S101_L003.crunchy.pending.txt does not exist
Compression/decompression is not running
Check if pending compression file exists
/home/proj/stage/housekeeper-bundles/ACC10629A3/2022-10-30/HMLN5DSX3_ACC10629A3_S101_L004.crunchy.pending.txt does not exist
Compression/decompression is not running
Dry run: Config files will not be written
Case richurchin exists in Status DB
Samples linked to case richurchin: 3
Case richurchin exists in Status DB
richurchin : analysis is still ongoing - skipping
Could not run analysis: Analysis still ongoing in Trailblazer for case richurchin
Unexpected error occurred:
Aborted!

File: cg/models/analysis.py [hunk 1]
```diff
@@ -9,6 +9,9 @@
 class AnalysisModel(BaseModel):
     """Metadata analysis model"""
 
+    class Config:
+        arbitrary_types_allowed = True
+
```
Comment (islean) [inline cg/models/analysis.py:14]: Why is this added? Is it needed?

File: cg/models/taxprofiler/taxprofiler.py [hunk 2]
```diff
@@ -40,7 +41,7 @@ class TaxprofilerParameters(WorkflowParameters):
     centrifuge_save_reads: bool = True
     run_krona: bool = True
     run_profile_standardisation: bool = True
-    clusterOptions: str = Field(..., alias="cluster_options")
+    clusterOptions: str | None = Field(str | None, alias="cluster_options")
```
Comment (islean) [inline cg/models/taxprofiler/taxprofiler.py:44]:
```suggestion
    clusterOptions: str | None = Field(None, alias="cluster_options")
```
I thought the first entry in "Field" is the default value but if not, please resolve. Why are we making this optional though? It seems required in the previous commit

File: cg/models/analysis.py (hunk 1 above)
Comment (henrikstranneheim) [reply cg/models/analysis.py:14]: Yes, due to the types in `class NextflowAnalysis(AnalysisModel):`

File: cg/models/taxprofiler/taxprofiler.py (hunk 2 above)
Comment (henrikstranneheim) [reply cg/models/taxprofiler/taxprofiler.py:44]: 👍

Comment (henrikstranneheim) [reply cg/models/taxprofiler/taxprofiler.py]: Fixed

File: cg/models/taxprofiler/taxprofiler.py [hunk 3]
```diff
@@ -40,7 +41,7 @@ class TaxprofilerParameters(WorkflowParameters):
     centrifuge_save_reads: bool = True
     run_krona: bool = True
     run_profile_standardisation: bool = True
-    clusterOptions: str = Field(..., alias="cluster_options")
+    clusterOptions: str = Field(str, alias="cluster_options")
```
Comment (islean) [inline cg/models/taxprofiler/taxprofiler.py:44]:
```suggestion
    clusterOptions: str = Field(alias="cluster_options")
```
I do not think you need the type hint in the Field instantiation at all

**Review by islean — APPROVED** [review]
Looks good!

**Comment by henrikstranneheim** [comment]
Deployed to prod
```
Log deploy... done.
cg, version 64.4.1
